### PR TITLE
Define an envelope type for newly added Ethereum tx.

### PR DESCRIPTION
**Pull request by sirano11**
## Proposed changes

This PR define an envelope type for newly added Ethereum tx.
 - To accept and distinguish Klaytn tx type and Ethereum tx type, it define an envelop type.
    - TxEthTypeEnvelope : 0x78
    - TxAccessList : 0x7801
    - TxDynamicFee : 0x7802
- In order to have the same value as Ethereum's txHash, TxTypeEthEnvelop is not used when creating a txHash.  

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

https://github.com/klaytn/klaytn/issues/1128

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/types/transaction_signing.go [hunk 1]
```diff
@@ -248,6 +248,107 @@ type Signer interface {
...
+
+// Hash returns the hash to be signed by the sender.
+// It does not uniquely identify the transaction.
+func (s londonSigner) Hash(tx *Transaction) common.Hash {
+	if tx.Type() != TxTypeDynamicFee {
+		return s.eip2930Signer.Hash(tx)
+	}
+
+	infs := append([]interface{}{s.ChainID()}, tx.data.SerializeForSign()...)
+	return prefixedRlpHash(byte(tx.Type()), infs)
```
Comment (aeharvlee) [inline blockchain/types/transaction_signing.go:343]:
```suggestion
	return prefixedRlpHash(byte(tx.Type()&0x0011), infs)
```
As we changed type from `0x1, 0x2` to `0x7801, 0x7802`
we need to do bit masking for this based on our SPEC.
```
// TxTypeAccessList (Klaytn Tx)
TxHashRLP = 0x01 || rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, signature_y_parity, signature_r, signature_s])
TxHash = keccak256(TxHashRLP)

// TxTypeDynamicFee (Klaytn Tx)
TxHashRLP = 0x02 || rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, amount, data, access_list, signature_y_parity, signature_r, signature_s]) 
TxHash = keccak256(TxHashRLP)
```

cc. @aidan-kwon @jimni1222

File: blockchain/types/tx_internal_data.go [hunk 2]
```diff
@@ -55,6 +55,9 @@ const (
 	TxTypeBatch, _, _
 	TxTypeChainDataAnchoring, TxTypeFeeDelegatedChainDataAnchoring, TxTypeFeeDelegatedChainDataAnchoringWithRatio
 	TxTypeLast, _, _
+	TxTypeEthEnvelope = 0x78
```
Comment (aeharvlee) [inline blockchain/types/tx_internal_data.go:58]:
```suggestion
	TxTypeEthEnvelope = TxType(0x78)
	TxTypeAccessList  = TxType(0x7801)
	TxTypeDynamicFee  = TxType(0x7802)
```

Without type conversion, they are not `TxType`, but just `int`.

File: blockchain/types/tx_internal_data.go [hunk 3]
```diff
@@ -55,6 +55,9 @@ const (
...
+	TxTypeAccessList = 0x7801
+	TxTypeDynamicFee = 0x7802
```
Comment (aeharvlee) [inline blockchain/types/tx_internal_data.go:60]:
```suggestion
	TxTypeEthEnvelope = TxType(0x78)
	TxTypeAccessList  = TxType(0x7801)
	TxTypeDynamicFee  = TxType(0x7802)
```

Without type conversion, they are not `TxType`, but just `int`.

File: blockchain/types/tx_internal_data_serializer.go [hunk 4]
```diff
@@ -50,9 +50,17 @@ func (serializer *TxInternalDataSerializer) EncodeRLP(w io.Writer) error {
 		return rlp.Encode(w, serializer.tx)
 	}
 
-	if err := rlp.Encode(w, serializer.txType); err != nil {
-		return err
+	if serializer.txType.IsEthTypedTransaction() {
+		ethType := uint8(serializer.txType)
+		if _, err := w.Write([]byte{TxTypeEthEnvelope, ethType}); err != nil {
```
Comment (aeharvlee) [inline blockchain/types/tx_internal_data_serializer.go:55]:
```suggestion
		if _, err := w.Write([]byte{byte(TxTypeEthEnvelope), ethType}); err != nil {
```

If we accept [the previous suggestion](https://github.com/klaytn/klaytn/pull/1164/files#r805378308), we need to convert it to byte type.

**Comment by sirano11** [comment]
It seems something wrong.... I close this PR and create New PR.

**Pull request closed without merging**